docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+  For example, suppose 90% of stake suddenly shut down their nodes, but somehow the remaining 10% was still well-connected and there remained a sufficiently dominant honest majority.
+  With just 10% of stake, it's almost certain that there would be less than k blocks in 3k/f slots, ie a Chain Growth violation.
+  However, it's also almost certain there would be hundreds of blocks in every window of 3k/f slots.
+  And, since the remaining nodes are still sufficiently well-connected, Praos's natural short forks would indeed remain short.
+  Thus R would remain much smaller than k, such that any node that needed to switch from D to C would still be able to validate R+1 headers after L on C despite not being able to predict more than 3k/f after the first header after L on C.
+
+## Back to Reality: the Cardano Node
+
+First, instead of merely pure Praos, consider a node running Cardano's hard fork combinator and its history of eras from Byron through Babbage.
+The answer to Question 1 actually does not change.
+One crucial reason is because Cardano sets the hard fork combinator safe zone of each era to that era's stability window.
+If the safe zone were instead less than one stability window, then L does not necessarily contain enough information to validate the R+1th header after L on C, _even without_ a Chain Growth violation.
+Specifically, the node will need to know if the chain will actually be in whichever era that header claims to be in, and it wouldn't necessarily be able to answer that question more than a safe zone after the first header after L on C.
+
+The answer to Question 2 does change: there is one additional change in behavior possibly enabled by a Chain Growth violation.
+Unlike the ledger, the hard fork combinator actually counts blocks instead of only slots.
+If a hypothetical roll back of k-many blocks could possibly change the ledger-based governance decision that was intended to trigger the era transition---which is true if Chain Growth was (severely) violated in the last two stability windows before the intended era transition---then the hard fork combinator will not transition to the next era.
+(For this reason, the voting deadline was moved from 7k/f to 4k/f (ie two stablity windows before the end) in the ledger governance rules, so that the hard fork combinator's final decision would be known at least a stability window (and so the safe zone) before the epoch transition.)
+Awkwardly, the ledger will still update the major version of the protocol version protocol parameter, but the chain will remain in the previous era, which many users and developers would find counter-intuitive.
+
+So this incoherence can arise only due to a Chain Growth violation (and a specific and severe one, at that), which means the hard fork combinator itself changes the answer to Question 2.
```

Comment:
I think we weren't yet aware of this when this text was written, but I think it would be good to mention that a less severe chain growth violation (when the k+1-st still occurs before the epoch end, but only in the last 3k/f slots) still has pretty weird behavior (https://github.com/input-output-hk/ouroboros-consensus/issues/385).